### PR TITLE
[SWPWA-228] Configurable product child product crosssells are displayed as well

### PR DESCRIPTION
**Pull request by edgars1337**

### CODE REVIEW

**Comment by Danpiel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/app/query/ProductList.query.js  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/base-theme/base-theme/pullRequest?prid=4587904&bid=15404910)

File: src/app/query/ProductList.query.js [hunk 1]
```diff
@@ -119,6 +119,7 @@ export class ProductListQuery {
 
     _getProductInterfaceFields(isVariant, getLinksFieldFromCart) {
```
Comment (alfredsgenkins) [inline src/app/query/ProductList.query.js]: Please make boolean variables start with `is`

File: src/app/query/ProductList.query.js [hunk 2]
```diff
@@ -130,6 +131,12 @@ export class ProductListQuery {
             this._getProductThumbnailField(),
             this._getShortDescriptionField(),
             this._getAttributesField(isVariant),
+            ...(isVariant && getLinksFieldFromCart
+                ? [
+                    this._getProductLinksField()
```
Comment (alfredsgenkins) [inline src/app/query/ProductList.query.js]: Looks like a mess. Please refactor. Why if `isVariant` and `getLinksFieldFromCart`? Why not just `getLinksFieldFromCart`?

File: src/app/store/Cart/Cart.dispatcher.js [hunk 3]
```diff
@@ -141,13 +141,16 @@ export class CartDispatcher {
...
         if (items.length > 0) {
             const product_links = items.reduce((links, product) => {
-                const { product: { product_links } } = product;
+                const { product: { product_links, variants = [] }, sku: variantSku } = product;
+                const { product_links: childProductLinks } = variants.find(({ sku }) => sku === variantSku);
+
+                if (childProductLinks) {
+                    Object.values(childProductLinks).filter(({ link_type }) => link_type === 'crosssell')
+                        .map(item => links.push(item));
+                }
```
Comment (alfredsgenkins) [inline src/app/store/Cart/Cart.dispatcher.js]: Enter here.

File: src/app/query/ProductList.query.js [hunk 4]
```diff
@@ -167,8 +174,9 @@ export class ProductListQuery {
     }
 
     _getProductField() {
+        const { getLinksFieldFromCart } = this.options;
         return new Field('product')
-            .addFieldList(this._getProductInterfaceFields(true));
+            .addFieldList(this._getProductInterfaceFields(true, getLinksFieldFromCart));
```
Comment (alfredsgenkins) [inline src/app/query/ProductList.query.js]: I do not like this logic it is too complicated! Why is this even related to query? Can you determine if you want this product links or not in reducer?
